test/oauth-server/service/authorize.go

```diff
@@ -93,5 +96,8 @@ func (requestHandler *RequestHandler) authorize(w http.ResponseWriter, r *http.R
 	resp := map[string]interface{}{
 		"code": code,
 	}
-	jsonResponseWriter(w, resp)
+	err = jsonResponseWriter(w, resp)
+	if err != nil {
+		writeError(w, err, http.StatusBadRequest)
```

Comment:
when w.Write fails you can only log error. it is not recommended to call twice w.WriteHeader